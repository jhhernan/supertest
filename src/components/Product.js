import React from 'react';

function Product(props) {
    const product = props.product;
    return (
        <div key={product.id} className="productBox" data-testid="product-box">
            <div className="interior">
                <div className="imageContainer">
                    <div className="image"><img src={product.thumbnail} alt="productimage" className="productImag" /></div>
                    <div className="sellosContainer">
                        {product.sellos && product.sellos.length > 0 && product.sellos.map((sello, idx) => (
                            <div className="tooltip1" key={idx}>
                                <img src={sello.image} alt="selloimagen" />
                                <span className="tooltiptext1">Producto<br />{sello.name.split("")[0].toUpperCase() + sello.name.slice(1)}</span>
                            </div>
                        ))}
                    </div>
                </div>
                <div className="productInfo">
                    <div className="supplierName">{product.supplier}</div>
                    <div className="weightInfo">{product.net_content}</div>
                    <p className="productTitle" data-testid="product-title">{product.title.length > 20 ? product.title.substr(0, 19) + "..." : product.title}</p>
                    <p className="productPrice"><span className="priceSymbol">$</span>{parseInt(product.price_real).toLocaleString()} <span className="productQuantity">x {product.units_sf} unids</span></p>
                </div>
            </div>
            <div className="addButton" data-testid="add-button" onClick={() => props.addCart(product)}>Agregar al carrito</div>
        </div>
    )
}

export default Product;