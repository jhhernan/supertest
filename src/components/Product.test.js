import React from 'react';
import { render, cleanup, fireEvent, waitForElement } from '@testing-library/react';
import Product from './Product';

const product = {
            "id": 446,
            "title": "Té Verde Pureza",
            "supplier": "Kundali",
            "tax": "0.19",
            "price_real": "22451",
            "image": "https://superfuds-file.s3.us-west-2.amazonaws.com/product/7709683463753_5e552d1f49f3a.png",
            "thumbnail": "https://superfuds-file.s3.us-west-2.amazonaws.com/product/thumbnail/7709683463753_5e552d1f49f3a.png",
            "description": "Tés e infusiones 100% natural alineado con los chakras. Combinamos experiencia gastronómica con afinidad bioenergética resultando en la combinación perfecta de suavidad + profundidad + sabor. Desintoxicación y limpieza a través de la esencia del cuatro chakra",
            "units_sf": 1,
            "slug": "te-verde-pureza",
            "category": "Bebidas",
            "subcategory": "Tés E Infusiones",
            "net_content": "30g",
            "sellos":[
                    {"name":"vegano","image":"https://s3-sa-east-1.amazonaws.com/assets.superfuds.com/assets/ecom/v3/Group.png"}
            ]
          }; 

describe('Product component test', () => {

    it('renders the component', () => {
        const container = render(<Product product={product} />)
        expect(container.firstChild).toMatchSnapshot()
    })

    it('display correctly using props data', () => {
        const { getAllByTestId, getByTestId } = render(
              <Product product={product} />
          );
        expect(getAllByTestId('product-title').length).toBe(1);
        expect(getByTestId('product-title').textContent).toBe("Té Verde Pureza");        
    })

    it('Calling addCart when clicking the button', () => {
      let addToCart = jest.fn();
      const { getAllByTestId, getByTestId } = render(
            <Product product={product} addCart={addToCart}/>
      );
      fireEvent.click(getByTestId('add-button'));
      expect(addToCart).toHaveBeenCalled();
    })


  });

