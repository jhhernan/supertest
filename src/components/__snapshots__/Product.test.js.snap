// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component test renders the component 1`] = `undefined`;
